arches/settings_utils.py

```diff
@@ -151,13 +165,19 @@ def build_templates_config(
         raise e
 
 
-def generate_frontend_configuration():
+def generate_frontend_configuration(app_config):
     try:
-        app_root_path = os.path.realpath(settings.APP_ROOT)
+        app_root_path = app_config.path
         root_dir_path = os.path.realpath(settings.ROOT_DIR)
 
-        arches_app_names = list_arches_app_names()
-        arches_app_paths = list_arches_app_paths()
+        if app_config.name == "arches":
+            app_root_path = os.path.join(app_config.path, "app")
+            base_path = root_dir_path
+        else:
+            base_path = app_root_path
+
+        arches_app_names = list_arches_app_names(application_name=app_config.name)
+        arches_app_paths = list_arches_app_paths(application_name=app_config.name)
```

Comment:
If we just use label, you don't have to split on `.` later.
```suggestion
        arches_app_names = list_arches_app_names(application_name=app_config.label)
        arches_app_paths = list_arches_app_paths(application_name=app_config.label)
```